enhancements/sig-architecture/32-extensiblescheduling/32-extensiblescheduling.md

```diff
@@ -0,0 +1,363 @@
+
+- Score
+
+In score stage, prioritizer plugin get valid scalar from ManagedClusterScalar CR. If no scalar updated in ManagedClusterScalar CR status, treat it as 0.
+After getting all the cluster scalar, the plugin should formalized them from -100 to 100. The maximal will be 100 and minimal is -100, other score distribute in proportion.
+
+- Re-Schedule
+
+Rather than watch ManagedClusterScalar and trigger Score(), that will cause frequently reschedule. The scheduler framework should periodically (eg. every 5 minutes) reschedule the placement.
+
+Future work：there might be a cluster-scoped CRD to configure the threshold of each prioritizer. For example, the frequency to trigger a re-scheudle, the threshold in prescore to treat score as valid.
+
+### Examples
+
+#### An agent update ResourceRatioCPU scalar to API, and placement prioritizer plugin score based on it. (Use Story 1&2)
+1. User create a new placement as below
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement
+  namespace: ns1
+spec:
+  numberOfClusters: 1
+  prioritizerPolicy:
+    mode: Exact
+    configurations:
+      - name: CustomizeResourceRatioCPU
+```
+2. Placement controller watch the placement changes and prioritizer plugin generate a new ManagedClusterScalar CR `resourceratiocpu` to trigger customized data collection.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterScalar
+metadata:
+  name: resourceratiocpu
+  namespace: cluster1
+spec:
+  prioritizerName: CustomizeResourceRatioCPU
+```
+3. The agent on managed cluster watch the new ManagedClusterScalar CR and update scalar into it. The agent refresh the scalar periodically before expire time.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterScalar
+metadata:
+  name: resourceratiocpu
+  namespace: cluster1
+spec:
+  prioritizerName: CustomizeResourceRatioCPU
+status:
+  conditions:
+  - lastTransitionTime: "2021-10-28T08:31:39Z"
+    message: ManagedClusterScalar updated successfully
+    reason: ManagedClusterScalarUpdated
+    status: "True"
+    type: ManagedClusterScalarUpdated
+  validUntil: "2021-10-29T18:31:39Z"
+  scalar: 74
+```
+4. Placement get scalar from ManagedClusterScalar CRs and sort clusters.
+Placement prioritizer plugin will wait for 80% of the ManagedClusterScalar CRs have scalar, then sort clusters and put the result in PlacementDecision. Placement will reschedule every 5 minutes to ensure the decision is up-to-date.
+
+#### Disaster recovery workload could be automatically switched to an avaliable cluster. (Use Story 3)
+1. A user has 2 clusters, one is primary and one is backup cluster. To deploy workloads on the primary cluster, define the placement as below.  
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement
+  namespace: ns1
+spec:
+  numberOfClusters: 1
```

Comment:
same suggetsion as above, the value of `numberOfClusters` should be bigger than 1

Reply:
This case is different from the above, as it's for the disaster recovery case and should always return only one cluster.